### PR TITLE
ADAPT-2679: Keyword Search Banners

### PR DESCRIPTION
**Pull request by andrewbaraniuk**
# READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- This PR creates a new SearchKeywordBanner component which shows Search Keyword Banner content below Search input in case entered keyword(s) match keyword(s) in Storyblok

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to path /search and enter some keyword, eg. "magazine" or "alumni", and hit Return
3. Verify that if some Search Keyword Banner with the same keyword(s) exist in Storyblok - it should be displayed below the search input field - https://prnt.sc/1bfgm6q

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/composite/searchKeywordBanner.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      })
+      let banner = '';
+      if (created) {
+        // Return most recently created Banner.
+        let max = Math.max(...Object.keys(created));
+        banner = created[max];
+      }
+      let content = '';
+      if (banner) {
+        switch (banner.content[0].component) {
```
Comment (nplowman) [inline src/components/composite/searchKeywordBanner.js:49]: I'd recommending using the `<CreateBloks>` component which handles this dynamically based on the mapping in components.js. That way if they add additional components to the whitelist, we don't need to adjust the code to support them.

File: src/components/composite/searchKeywordBanner.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          created_at
+          field_keywords_string
+        }
+      }
+    }
+  }
+`;
+
+const SearchKeywordBanner = ({ queryText }) => (
+  <StaticQuery
```
Comment (nplowman) [inline src/components/composite/searchKeywordBanner.js:28]: I'd recommend moving the preprocess logic here where you're checking for the keyword matching into a separate function - just a little tidier than having it all in the JSX template. (Probably fine to keep in the same file).

Also wanted to make sure you were aware that there is also a hook for static query that you can use if you prefer that syntax over the component-based version:
https://www.gatsbyjs.com/docs/how-to/querying-data/use-static-query/

(Mostly just a personal preference, but I like it a little better)

**Review by nplowman — CHANGES_REQUESTED** [review]
Overall this is looking good. Just had a couple of suggestions.

File: src/components/page-types/searchPage.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ import SearchResults from "../search/searchResults";
 import SearchPager from "../search/searchPager";
 import SearchFacet from "../search/searchFacet";
 import SearchNoResults from "../search/searchNoResults";
+import SearchKeywordBanner from "../composite/searchKeywordBanner";
```
Comment (sherakama) [inline src/components/page-types/searchPage.js:18]: Please move this template into the `search` directory.

Comment (sherakama) [reply src/components/page-types/searchPage.js:18]:
```suggestion
import SearchKeywordBanner from "../search/searchKeywordBanner";
```

File: src/components/composite/searchKeywordBanner.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    const split = node.field_keywords_string.split(",");
+    const newSplit = split.map((str) => str.trim());
+    if (newSplit.indexOf(q) >= 0) {
+      created[Date.parse(node.created_at)] = blok;
+    }
+
+    return created;
+  });
+
+  return created ? created[Math.max(...Object.keys(created))].content : "";
```
Comment (sherakama) [inline src/components/composite/searchKeywordBanner.js:19]: I updated the branch with the code from main and published the two banners in storyblok and now this line is throwing an error in the build. Looks like it isn't finding any content. Can you refactor this please.

**Review by sherakama — CHANGES_REQUESTED** [review]
When the build was working the placement of the banners and functionality joined with the search was good.

**Review by sherakama — APPROVED** [review]
OK.

**Pull request merged**